### ISSUE TITLE
[US] Failed authentication

### ISSUE THREAD

**Issue by athal7** [Hacksore/bluelinky#155]
**Describe the bug**
As of the last few days, there are error responses coming from the Hyundai API on attempted auth. Seems unrelated to any release.

**Useful info(please complete the following information):**
 - OS: Linux
 - Region: US

**Additional context**

https://github.com/athal7/homebridge-hyundai-bluelink/issues/26

**Comment by Hacksore** [comment]
Looks like we might have a new client id or something related to the new app.

```json
{
  "errorSubCode": "C401",
  "errorSubMessage": "An updated version of this app is now available. Please download the latest version.",
  "errorMessage": "An updated version of this app is now available. Please download the latest version.",
  "errorCode": 400
}

 ```

**Comment by PierreLevres** [comment]
Striking, but that is the exact same message I get when running the old Kia-UVO iPhone app, but that is another region (EU).

**Comment by Hacksore** [comment]
Was able to get the latest client id/secret for the US API using this frida script.

client_id: `m66129Bb-em93-SPAHYN-bZ91-am4540zp19920`
client_secret: `v558o935-6nne-423i-baa8`

```js
// Root bypass
Java.perform(function () {
  const Login = Java.use(
    "com.stationdm.bluelink.kotlin.ui.auth.SDMSignInActivity"
  );
  const Intent = Java.use("android.content.Intent");

  const Main = Java.use(
    "com.stationdm.bluelink.kotlin.ui.auth.SDMSplashActivity"
  );
  Main.onCreate.overload("android.os.Bundle").implementation = function (
    bundle
  ) {
    this.onCreate(bundle);
    console.log("\nInside the SDMSplashActivity function");

    const myIntent = Intent.$new(this, Login.class);

    this.startActivity(myIntent);
    return;
  };
});

// ssl pin
Java.perform(function () {
  console.log("[#] Android Bypass for various Certificate Pinning methods [#]");

  const X509TrustManager = Java.use("javax.net.ssl.X509TrustManager");
  const SSLContext = Java.use("javax.net.ssl.SSLContext");

  // TrustManager (Android < 7)
  const TrustManager = Java.registerClass({
    // Implement a custom TrustManager
    name: "dev.asd.test.TrustManager",
    implements: [X509TrustManager],
    methods: {
      checkClientTrusted: function (chain, authType) {},
      checkServerTrusted: function (chain, authType) {},
      getAcceptedIssuers: function () {
        return [];
      },
    },
  });

  // Prepare the TrustManager array to pass to SSLContext.init()
  const TrustManagers = [TrustManager.$new()];
  // Get a handle on the init() on the SSLContext class
  const SSLContext_init = SSLContext.init.overload(
    "[Ljavax.net.ssl.KeyManager;",
    "[Ljavax.net.ssl.TrustManager;",
    "java.security.SecureRandom"
  );
  try {
    // Override the init method, specifying the custom TrustManager
    SSLContext_init.implementation = function (
      keyManager,
      trustManager,
      secureRandom
    ) {
      console.log("[+] Bypassing Trustmanager (Android < 7) request");
      SSLContext_init.call(this, keyManager, TrustManagers, secureRandom);
    };
  } catch (err) {
    console.log("[-] TrustManager (Android < 7) pinner not found");
    //console.log(err);
  }

  // OkHTTPv3 (double bypass)
  try {
    const okhttp3_Activity = Java.use("okhttp3.CertificatePinner");
    okhttp3_Activity.check.overload(
      "java.lang.String",
      "java.util.List"
    ).implementation = function (str) {
      console.log("[+] Bypassing OkHTTPv3 {1}: " + str);
      return true;
    };
    // This method of CertificatePinner.check could be found in some old Android app
    okhttp3_Activity.check.overload(
      "java.lang.String",
      "java.security.cert.Certificate"
    ).implementation = function (str) {
      console.log("[+] Bypassing OkHTTPv3 {2}: " + str);
      return true;
    };
  } catch (err) {
    console.log("[-] OkHTTPv3 pinner not found");
    //console.log(err);
  }
});
```

**Comment by Hacksore** [comment]
Fixed via #156
